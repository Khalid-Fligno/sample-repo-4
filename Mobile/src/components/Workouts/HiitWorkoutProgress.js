import React from "react";
import { View, StyleSheet, Text } from "react-native";
import PropTypes from "prop-types";
import { PieChart } from "react-native-svg-charts";
import Icon from "../Shared/Icon";
import colors from "../../styles/colors";
import fonts from "../../styles/fonts";

const HiitWorkoutProgress = (props) => {
  const { currentRound, rest, rounds } = props;
  const array = new Array(rounds).fill(undefined).map((val, idx) => idx);
  const pieDataIncomplete = [0, 100].map((value, index) => ({
    value,
    svg: {
      fill: index === 0 ? colors.coral.standard : colors.grey.light,
    },
    key: `pie-${index}`,
  }));

  const pieDataSetHalf = [50, 50].map((value, index) => ({
    value,
    svg: {
      fill: index === 0 ? colors.grey.light : colors.coral.standard,
    },
    key: `pie-${index}`,
  }));

  const pieDataComplete = [100, 0].map((value, index) => ({
    value,
    svg: {
      fill: colors.coral.standard,
    },
    key: `pie-${index}`,
  }));

  const completePieChart = (
    <PieChart
      style={styles.pieChart}
      data={pieDataComplete}
      innerRadius="80%"
    />
  );
  const tickIcon = (
    <View style={styles.invisibleView}>
      <View style={styles.tickContainer}>
        <Icon name="tick-heavy" color={colors.charcoal.dark} size={18} />
      </View>
    </View>
  );
  const currentRoundText = (text) => (
    <View style={styles.invisibleView}>
      <View style={styles.currentRoundTextContainer}>
        <Text style={styles.currentRoundText}>{text}</Text>
      </View>
    </View>
  );
  const currentRoundPieChart = (resting) => (
    <PieChart
      style={styles.pieChart}
      data={resting ? pieDataSetHalf : pieDataIncomplete}
      innerRadius="80%"
      startAngle={Math.PI * 2}
      endAngle={0}
    />
  );
  const incompletePieChart = (
    <PieChart
      style={styles.pieChart}
      data={pieDataIncomplete}
      innerRadius="80%"
    />
  );
  const inactiveRoundText = (text) => (
    <View style={styles.invisibleView}>
      <View style={styles.inactiveRoundTextContainer}>
        <Text style={styles.inactiveRoundText}>{text}</Text>
      </View>
    </View>
  );

  return (
    <View style={styles.container}>
      {array.map((item, index) => (
        <View style={styles.exercise} key={index}>
          {currentRound > index + 1 && completePieChart}
          {currentRound > index + 1 && tickIcon}
          {currentRound === index + 1 && currentRoundPieChart(rest)}
          {currentRound === index + 1 && currentRoundText(currentRound)}
          {currentRound < index + 1 && incompletePieChart}
          {currentRound < index + 1 && inactiveRoundText(index + 1)}
        </View>
      ))}
    </View>
  );
};

HiitWorkoutProgress.propTypes = {
  currentRound: PropTypes.number.isRequired,
  rest: PropTypes.bool,
};

HiitWorkoutProgress.defaultProps = {
  rest: false,
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    height: 35,
    paddingLeft: 5,
    paddingRight: 5,
    marginTop: 10,
    marginBottom: 10,
  },
  exercise: {
    flex: 1,
    paddingLeft: 5,
    paddingRight: 5,
    alignItems: "center",
    justifyContent: "center",
  },
  pieChart: {
    height: 35,
    width: 35,
  },
  invisibleView: {
    height: 0,
  },
  tickContainer: {
    marginTop: -26,
  },
  currentRoundTextContainer: {
    marginTop: -26,
  },
  currentRoundText: {
    fontFamily: fonts.bold,
    fontSize: 18,
    color: colors.charcoal.dark,
  },
  inactiveRoundTextContainer: {
    marginTop: -26,
  },
  inactiveRoundText: {
    fontFamily: fonts.bold,
    fontSize: 18,
    color: colors.grey.light,
  },
});

export default HiitWorkoutProgress;
