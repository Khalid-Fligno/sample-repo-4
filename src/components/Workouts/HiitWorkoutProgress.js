import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import PropTypes from 'prop-types';
import { PieChart } from 'react-native-svg-charts';
import Icon from '../Icon';
import colors from '../../styles/colors';
import fonts from '../../styles/fonts';

const pieDataIncomplete = [0, 100]
  .map((value, index) => ({
    value,
    svg: {
      fill: index === 0 ? colors.coral.standard : colors.grey.light,
    },
    key: `pie-${index}`,
  }));

const pieDataSetHalf = [50, 50]
  .map((value, index) => ({
    value,
    svg: {
      fill: index === 0 ? colors.grey.light : colors.coral.standard,
    },
    key: `pie-${index}`,
  }));

const pieDataComplete = [100, 0]
  .map((value, index) => ({
    value,
    svg: {
      fill: colors.coral.standard,
    },
    key: `pie-${index}`,
  }));

class HiitWorkoutProgress extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
    };
  }
  render() {
    const { currentRound, rest } = this.props;
    const completePieChart = (
      <PieChart
        style={styles.pieChart}
        data={pieDataComplete}
        innerRadius="80%"
      />
    );
    const tickIcon = (
      <View style={styles.invisibleView}>
        <View style={styles.tickContainer}>
          <Icon
            name="tick-heavy"
            color={colors.charcoal.dark}
            size={18}
          />
        </View>
      </View>
    );
    const currentRoundText = (text) => (
      <View style={styles.invisibleView}>
        <View style={styles.currentRoundTextContainer}>
          <Text style={styles.currentRoundText}>
            {text}
          </Text>
        </View>
      </View>
    );
    const currentRoundPieChart = (resting) => (
      <PieChart
        style={styles.pieChart}
        data={resting ? pieDataSetHalf : pieDataIncomplete}
        innerRadius="80%"
        startAngle={Math.PI * 2}
        endAngle={0}
      />
    );
    const incompletePieChart = (
      <PieChart
        style={styles.pieChart}
        data={pieDataIncomplete}
        innerRadius="80%"
      />
    );
    const inactiveRoundText = (text) => (
      <View style={styles.invisibleView}>
        <View style={styles.inactiveRoundTextContainer}>
          <Text style={styles.inactiveRoundText}>
            {text}
          </Text>
        </View>
      </View>
    );
    return (
      <View style={styles.container}>
        <View style={styles.exercise}>
          {currentRound > 1 && completePieChart}
          {currentRound > 1 && tickIcon}
          {currentRound === 1 && currentRoundPieChart(rest)}
          {currentRound === 1 && currentRoundText(currentRound)}
        </View>
        <View style={styles.exercise}>
          {currentRound > 2 && completePieChart}
          {currentRound > 2 && tickIcon}
          {currentRound === 2 && currentRoundPieChart(rest)}
          {currentRound === 2 && currentRoundText(currentRound)}
          {currentRound < 2 && incompletePieChart}
          {currentRound < 2 && inactiveRoundText(2)}
        </View>
        <View style={styles.exercise}>
          {currentRound > 3 && completePieChart}
          {currentRound > 3 && tickIcon}
          {currentRound === 3 && currentRoundPieChart(rest)}
          {currentRound === 3 && currentRoundText(currentRound)}
          {currentRound < 3 && incompletePieChart}
          {currentRound < 3 && inactiveRoundText(3)}
        </View>
        <View style={styles.exercise}>
          {currentRound > 4 && completePieChart}
          {currentRound > 4 && tickIcon}
          {currentRound === 4 && currentRoundPieChart(rest)}
          {currentRound === 4 && currentRoundText(currentRound)}
          {currentRound < 4 && incompletePieChart}
          {currentRound < 4 && inactiveRoundText(4)}
        </View>
        <View style={styles.exercise}>
          {currentRound > 5 && completePieChart}
          {currentRound > 5 && tickIcon}
          {currentRound === 5 && currentRoundPieChart(rest)}
          {currentRound === 5 && currentRoundText(currentRound)}
          {currentRound < 5 && incompletePieChart}
          {currentRound < 5 && inactiveRoundText(5)}
        </View>
        <View style={styles.exercise}>
          {currentRound > 6 && completePieChart}
          {currentRound > 6 && tickIcon}
          {currentRound === 6 && currentRoundPieChart(rest)}
          {currentRound === 6 && currentRoundText(currentRound)}
          {currentRound < 6 && incompletePieChart}
          {currentRound < 6 && inactiveRoundText(6)}
        </View>
        <View style={styles.exercise}>
          {currentRound > 7 && completePieChart}
          {currentRound > 7 && tickIcon}
          {currentRound === 7 && currentRoundPieChart(rest)}
          {currentRound === 7 && currentRoundText(currentRound)}
          {currentRound < 7 && incompletePieChart}
          {currentRound < 7 && inactiveRoundText(7)}
        </View>
        <View style={styles.exercise}>
          {currentRound > 8 && completePieChart}
          {currentRound > 8 && tickIcon}
          {currentRound === 8 && currentRoundPieChart(rest)}
          {currentRound === 8 && currentRoundText(currentRound)}
          {currentRound < 8 && incompletePieChart}
          {currentRound < 8 && inactiveRoundText(8)}
        </View>
      </View>
    );
  }
}

HiitWorkoutProgress.propTypes = {
  currentRound: PropTypes.number.isRequired,
  rest: PropTypes.bool,
};

HiitWorkoutProgress.defaultProps = {
  rest: false,
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    height: 35,
    paddingLeft: 10,
    paddingRight: 10,
    marginTop: 10,
    marginBottom: 10,
  },
  exercise: {
    flex: 1,
    paddingLeft: 5,
    paddingRight: 5,
    alignItems: 'center',
    justifyContent: 'center',
  },
  pieChart: {
    height: 35,
    width: 35,
  },
  invisibleView: {
    height: 0,
  },
  tickContainer: {
    marginTop: -26,
  },
  currentRoundTextContainer: {
    marginTop: -26,
  },
  currentRoundText: {
    fontFamily: fonts.bold,
    fontSize: 18,
    color: colors.charcoal.dark,
  },
  inactiveRoundTextContainer: {
    marginTop: -26,
  },
  inactiveRoundText: {
    fontFamily: fonts.bold,
    fontSize: 18,
    color: colors.grey.light,
  },
});

export default HiitWorkoutProgress;
