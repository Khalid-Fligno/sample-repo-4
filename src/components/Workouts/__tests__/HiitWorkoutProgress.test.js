import React from 'react';
import renderer from 'react-test-renderer';
import HiitWorkoutProgress from '../HiitWorkoutProgress';

describe('Hiit Workout Progress', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HiitWorkoutProgress
      currentRound={1}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
