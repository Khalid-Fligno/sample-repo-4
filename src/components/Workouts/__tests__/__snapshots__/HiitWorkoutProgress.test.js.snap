// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hiit Workout Progress renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 35,
      "marginBottom": 10,
      "marginTop": 10,
      "paddingLeft": 5,
      "paddingRight": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    <View
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": -26,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(35, 36, 44)",
              "fontFamily": "GothamBold",
              "fontSize": 18,
            }
          }
        >
          1
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    <View
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": -26,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(234, 234, 235)",
              "fontFamily": "GothamBold",
              "fontSize": 18,
            }
          }
        >
          2
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    <View
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": -26,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(234, 234, 235)",
              "fontFamily": "GothamBold",
              "fontSize": 18,
            }
          }
        >
          3
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    <View
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": -26,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(234, 234, 235)",
              "fontFamily": "GothamBold",
              "fontSize": 18,
            }
          }
        >
          4
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    <View
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": -26,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(234, 234, 235)",
              "fontFamily": "GothamBold",
              "fontSize": 18,
            }
          }
        >
          5
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    <View
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": -26,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(234, 234, 235)",
              "fontFamily": "GothamBold",
              "fontSize": 18,
            }
          }
        >
          6
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    <View
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": -26,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(234, 234, 235)",
              "fontFamily": "GothamBold",
              "fontSize": 18,
            }
          }
        >
          7
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    <View
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": -26,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(234, 234, 235)",
              "fontFamily": "GothamBold",
              "fontSize": 18,
            }
          }
        >
          8
        </Text>
      </View>
    </View>
  </View>
</View>
`;
